src/etc/inc/pkg-utils.inc

```diff
@@ -359,21 +361,72 @@ function get_pkg_info($pkgs = 'all', $only_local = false) {
 
 	/* Do not run remote operations if pkg has a lock */
 	if (is_subsystem_dirty('pkg')) {
-		$only_local = true;
+		$local_only = true;
 		$lock = false;
 	} else {
 		$lock = true;
 	}
 
 	$extra_param = "";
-	if ($only_local) {
+	if ($local_only) {
 		$extra_param = "-U ";
 	}
 
 	if ($lock) {
 		mark_subsystem_dirty('pkg');
 	}
-	$rc = pkg_exec("search {$extra_param}-R --raw-format json-compact " . $pkgs, $out, $err);
+
+	if (!$installed_only) {
+		// repo catalog search (either remote or local_only)
+		$rc = pkg_exec("search {$extra_param}-R --raw-format json-compact " . $pkgs, $out, $err);
+	}
+	if ($installed_only || ($local_only && $rc != 0)) {
+		/* use pkg info if (1) installed pkg search or (2) local catalog copy search requested + failed.
+		 *
+		 * The local repo catalog copy may be cleared if a previous call to pkg search couldn't get the
+		 * remote repo catalog.
+		 *
+		 * If the calling code would have accepted local copy info (which isn't assumed up to date) then it
+		 * makes sense to fall back on pkg info to at least return the known info about installed pkgs (pkg
+		 * info should still work), instead of failing and returning no info at all. For example, this
+		 * enables offline view + management of installed pkgs.
+		 */
+
+		// pkg info errors if none match, unlike pkg search, so test it will work beforehand...
+		if (
+
+	$extra_param = "";
+	if ($local_only) {
+		$extra_param = "-U ";
+	}
+
+	if ($lock) {
+		mark_subsystem_dirty('pkg');
+	}
+
+	if (!$installed_only) {
+		// repo catalog search (either remote or local_only)
+		$rc = pkg_exec("search {$extra_param}-R --raw-format json-compact " . $pkgs, $out, $err);
+	}
+	if ($installed_only || ($local_only && $rc != 0)) {
+		/* use pkg info if (1) installed pkg search or (2) local catalog copy search requested + failed.
+		 *
+		 * The local repo catalog copy may be cleared if a previous call to pkg search couldn't get the
+		 * remote repo catalog.
+		 *
+		 * If the calling code would have accepted local copy info (which isn't assumed up to date) then it
+		 * makes sense to fall back on pkg info to at least return the known info about installed pkgs (pkg
+		 * info should still work), instead of failing and returning no info at all. For example, this
+		 * enables offline view + management of installed pkgs.
+		 */
+
+		// pkg info errors if none match, unlike pkg search, so test it will work beforehand
+		// is_package_installed() is a wrapper for pkg info -e <pattern> which is what we need here.
+		if (is_package_installed($pkgs)) {
+			// ok, 1 or more packages match, so pkg info can be safely called to get the pkg list
+			$rc = pkg_exec("info -R --raw-format json-compact " . $pkgs, $out, $err);
+		} // else we already have empty values for $out etc which are correct if none matched
+	}
```

Comment:
I don't think that's leftover. Remember that the function get_pkg_info() returns data for the packages matching `pattern`. The point of the PR is that if a remote repo can't be accessed and/or local installed pkg info would be enough, at least we can fall back on getting local installed pkg info by using pkg info not pkg search. But you still have to actually execute the pkg info call to get that info (in those circumstances and once you've checked it won't error out by pre-calling with the -e option).